usaspending_api/awards/delta_models/financial_accounts_by_awards.py

```diff
@@ -62,3 +62,308 @@
+            aw.id AS award_id,
+            5 AS priority
+        FROM
+            faba_upper AS faba
+        INNER JOIN
+            awards_upper AS aw
+        ON
+            aw.fain = faba.fain
+        WHERE
+            faba.fain IS NOT NULL
+            AND faba.uri IS NOT NULL
+            AND faba.award_id IS NULL
+            AND (
+                SELECT COUNT(*)
+                FROM awards_upper AS aw_sub
+                WHERE
+                    aw_sub.fain = faba.fain
+            ) = 1
+    );
+    """,
+    # -----
+    # When both FAIN and URI are populated in File C, update File C assistance records to link to a
+    # corresponding award if there is an single exact match based on URI
+    # -----
+    """
+    CREATE OR REPLACE TEMPORARY VIEW c_to_d_fain_and_uri_uri AS (
+        SELECT
+            financial_accounts_by_awards_id,
+            aw.id AS award_id,
+            6 AS priority
+        FROM
+            faba_upper AS faba
+        INNER JOIN
+            awards_upper AS aw
+        ON
+            aw.uri = faba.uri
+        WHERE
+            faba.fain IS NOT NULL
+            AND faba.uri IS NOT NULL
+            AND faba.award_id IS NULL
+            AND (
+                SELECT COUNT(*)
+                FROM awards_upper AS aw_sub
+                WHERE
+                    aw_sub.uri = faba.uri
+            ) = 1
+    );
+    """,
+    # -----
+    # Determine which FABA records need to be unlinked to an Award because its corresponding Award
+    # was deleted;
+    # This is necessary in addition to the first SQL statement checking for deletes. The first view
+    # was of ALL FABA records and included NULLs on deleted awards, so those FABA records would
+    # be included in the previous six `c_to_d_*` linkage views. This second view checking for deletes
+    # is adding a new `c_to_d_*` linkage view to be included in the final UNION and MERGE into the
+    # `int` version of `financial_accounts_by_awards`
+    # -----
+    """
+    CREATE OR REPLACE TEMPORARY VIEW c_to_d_deletes AS (
+        SELECT
+            faba.financial_accounts_by_awards_id,
+            aw.id,
+            7 AS priority
+        FROM
+            int.financial_accounts_by_awards AS faba
+        LEFT JOIN
+            awards_upper AS aw
+        ON
+            faba.award_id = aw.id
+        WHERE
+            aw.id IS NULL
+            AND faba.award_id IS NOT NULL
+    );
+    """,
+    # -----
+    # Union all the `c_to_d_*` linkage views into a single view
+    # -----
+    """
+    CREATE OR REPLACE TEMPORARY VIEW union_all AS (
+        SELECT * FROM c_to_d_piid_and_parent_piid
+        UNION
+        SELECT * FROM c_to_d_piid
+        UNION
+        SELECT * FROM c_to_d_fain
+        UNION
+        SELECT * FROM c_to_d_uri
+        UNION
+        SELECT * FROM c_to_d_fain_and_uri_fain
+        UNION
+        SELECT * FROM c_to_d_fain_and_uri_uri
+        UNION
+        SELECT * FROM c_to_d_deletes
```

Comment:
Since each temp view will have a different priority, `UNION` will never collapse down duplicate records. Can a `UNION ALL` then be used, since it typically is more efficient (no record comparisons needed)?